### PR TITLE
702 Move processSteps query to RTK

### PR DESCRIPTION
**Pull request by DutchBen**

### CODE REVIEW

File: packages/orchestrator-ui-components/src/rtk/endpoints/processSteps.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    endpoints: (build) => ({
+        getTimeLineItems: build.query<TimelineItem[], Process['workflowName']>({
+            query: (processName) => ({
+                document: processStepsQuery,
+                variables: { processName },
+            }),
+            transformResponse: (
+                response: ProcessStepsResult,
+            ): TimelineItem[] => {
+                const timeLineItems: TimelineItem[] = response.workflows
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/rtk/endpoints/processSteps.ts:34]: According to the type definition the workflows prop (ProcessStepsResult) is always available, there for there is no need for using `?` here. Same for page[0].

File: packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx [hunk 2]
```diff
@@ -97,30 +96,15 @@ export const WfoStartProcessPage = ({
...
+        useGetTimeLineItemsQuery(processName);
 
     if (isError) {
         if (!hasError) {
             setHasError(true);
         }
     }
 
-    const timeLineItems: TimelineItem[] =
-        data?.workflows?.page[0]?.steps && !isLoading
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx:115]: Probably slightly related to my comment in `packages/orchestrator-ui-components/src/rtk/endpoints/processSteps.ts`, you might want to act on isLoading from the useGetTimeLineItemsQuery hook

**Comment by ricardovdheijden** [comment]
> ### ⚠️ No Changeset found
> Latest commit: [617551b](https://github.com/workfloworchestrator/orchestrator-ui-library/commit/617551b35ae10d2a280db44d67c15c91d430437f)
> 
> Merging this PR will not cause a version bump for any packages. If these changes should not result in a new version, you're good to go. **If these changes should result in a version bump, you need to add a changeset.**
> 
> This PR includes no changesets
> [Click here to learn what changesets are, and how to add one](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md).
> 
> [Click here if you're a maintainer who wants to add a changeset to this PR](https://github.com/workfloworchestrator/orchestrator-ui-library/new/702-rtk-process?filename=.changeset/violet-peas-visit.md&value=---%0A%22%40orchestrator-ui%2Forchestrator-ui-components%22%3A%20patch%0A---%0A%0A702%20Move%20processSteps%20query%20to%20RTK%0A)

@DutchBen , also add a changeset file for this feature

File: packages/orchestrator-ui-components/src/rtk/endpoints/processSteps.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                    assignee
+                }
+            }
+        }
+    }
+`;
+
+const processStepApi = orchestratorApi.injectEndpoints({
+    endpoints: (build) => ({
+        getTimeLineItems: build.query<TimelineItem[], Process['workflowName']>({
```
[the remark this answers is not in the document]
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/rtk/endpoints/processSteps.ts:26]: For this case no cachekey is needed. Cache invalidation for these values will never happen

File: packages/orchestrator-ui-components/src/rtk/endpoints/processSteps.ts (hunk 1 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/rtk/endpoints/processSteps.ts:34]: Fixed

File: packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx (hunk 2 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/pages/processes/WfoStartProcessPage.tsx:115]: Fixed

**Review by ricardovdheijden — APPROVED** [review]

**Pull request merged**